app/models/concerns/auditable.rb

```diff
@@ -8,12 +8,21 @@ module Auditable
 
     def audit!(activity_type:, activity_information: nil, audit_comment: nil)
       audits.create!(
-        user: Current.user,
+        user: current_user,
         activity_type: activity_type,
         activity_information: activity_information,
         audit_comment: audit_comment,
-        api_user: Current.api_user
+        api_user: current_api_user,
+        automated_activity: current_user.blank? && current_api_user.blank?
```

Comment:
At this stage if there's no current user or api user we know it's an automated task!

Reply:
How about moving it into a method `#has_current_api_and_user?` or any other naming you prefer?

Reply:
Good shout!